### ISSUE TITLE
Enhancement: programmatic hook, or structured output, for downstream CLI tools

### ISSUE THREAD

**Issue by BatmanAoD** [codingjoe/relint#108]
I'd like to be able to generate "possible" lints with `relint`, and then use other CLI tools to determine whether each lint is a false positive or not.

Specifically, my use-case is to find instances of `\bTODO #\d+` and check whether the `#\d+` refers to an _open_ ticket. This requires checking the status of tickets in a ticket-tracker, which is well outside the purview of `relint`. But if `relint` had a way to feed a particular lint to a downstream program, especially using regex captures, this would be pretty simple:

```yaml
- name: TODOS with tickets
  pattern: '\bTODO #(\d+)'
  hint: make sure that the ticket is open
  error: false
  filePattern: '^(?!docs/)'
  postprocess: "glab issue show '${1}' | grep '^state:' | grep -q open"
```

Here, a status code of `1` means that the lint IS valid, and `0` means it's a false positive. (I'm not sure whether there's a better way to do this than using status codes, and it's not obvious whether `0` should mean "valid" or "false positive", but I think those are primarily documentation issues.)

**Comment by mroy-seedbox** [comment]
Related:
- [Secondary regex matching step](https://github.com/codingjoe/relint/issues/15#issuecomment-2403968281) (does not fulfil the requirements described above).
- [Test functions in Python](https://github.com/codingjoe/relint/issues/15#issuecomment-2404539424) (could be used to address the requirements described above).

Bonus: if `test_func` is used in combination with `pattern`, it should only operate on regex matches instead of on the entire file. This would make it the equivalent of the `postprocess` proposed above, but would make for a better integration than CLI exit codes (assuming that some kind of Python interface would exist in order to integrate with the `test_func`s with reLint).
